packages/type-utils/src/isTypeReadonly.ts

```diff
@@ -157,6 +157,12 @@ function isTypeReadonlyObject(
         continue;
       }
 
+      const name = ts.getNameOfDeclaration(property.valueDeclaration);
+      const isPrivateIdentifier = name && ts.isPrivateIdentifier(name);
+      if (isPrivateIdentifier) {
+        continue;
+      }
+
```

Comment:
It seems like there is no handy function for checking `private identifier` from ts or tsutils. I borrowed from a [typescript's src/compiler/checker.ts](https://raw.githubusercontent.com/microsoft/TypeScript/main/src/compiler/checker.ts) that suits this situation.

```
 function checkIndexConstraintForProperty(type: Type, prop: Symbol, propNameType: Type, propType: Type) {
    const declaration = prop.valueDeclaration;
    const name = getNameOfDeclaration(declaration);
    if (name && isPrivateIdentifier(name)) {
        return;
    }
    ........
}
```